ambry-store/src/main/java/com/github/ambry/store/CompactionLog.java

```diff
@@ -72,6 +81,60 @@ static boolean isCompactionInProgress(String dir, String storeId) {
     return new File(dir, storeId + COMPACTION_LOG_SUFFIX).exists();
   }
 
+  /**
+   * The {@link CompactionLog} processor.
+   */
+  @FunctionalInterface
+  interface CompactionLogProcessor {
+
+    /**
+     * Process a given {@link CompactionLog}.
+     * @param log The {@link CompactionLog} to process.
+     * @return True if you want to process more compaction log.
+     */
+    boolean process(CompactionLog log);
+  }
+
+  /**
+   * Process all the available {@link CompactionLog}s in descending order. The CompactionLog created most recently would
+   * be processed first.
+   * @param dir The data dir that contains all the compaction log files.
+   * @param storeId The store id.
+   * @param storeKeyFactory The {@link StoreKeyFactory}.
+   * @param time The time instance.
+   * @param config The {@link StoreConfig} object.
+   * @param processor The {@link CompactionLogProcessor}.
+   * @throws IOException
+   */
+  static void processCompactionLogs(String dir, String storeId, StoreKeyFactory storeKeyFactory, Time time,
+      StoreConfig config, CompactionLogProcessor processor) throws IOException {
+    // Ignore current in process compaction.
+    File storeDir = new File(dir);
+    File[] files =
+        storeDir.listFiles((fileDir, name) -> name.startsWith(storeId + COMPACTION_LOG_SUFFIX + BlobStore.SEPARATOR));
+    List<File> sortedFiles = Stream.of(files)
+        .sorted((file1, file2) -> (int) (getStartTimeFromFile(file2) - getStartTimeFromFile(file1)))
+        .collect(Collectors.toList());
+    File inProgressCompactionLog = new File(dir, storeId + COMPACTION_LOG_SUFFIX);
+    if (inProgressCompactionLog.exists()) {
+      sortedFiles.add(0, inProgressCompactionLog); // If there is an in progress compaction log, put it in the front.
+    }
+    for (File file : sortedFiles) {
+      if (!processor.process(new CompactionLog(file, storeKeyFactory, time, config))) {
+        break;
+      }
+    }
+  }
+
+  /**
+   * Getting start time from the compaction log file.
+   * @param file The compaction log file.
+   * @return The start time in milliseconds.
+   */
+  static long getStartTimeFromFile(File file) {
+    return Long.valueOf(file.getName().split(BlobStore.SEPARATOR)[2]);
```

Comment:
nit: you could use Long.parse since you just need a primitive instead of object right?